playground/AzureContainerApps/AzureContainerApps.AppHost/infra.module.bicep

```diff
@@ -0,0 +1,173 @@
+@description('The location for the resource(s) to be deployed.')
+param location string = resourceGroup().location
+
+param principalId string
+
+param tags object = { }
+
+resource mi 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' = {
+  name: take('mi-${uniqueString(resourceGroup().id)}', 128)
+  location: location
+  tags: tags
+}
+
+resource acr 'Microsoft.ContainerRegistry/registries@2023-07-01' = {
+  name: take('acr${uniqueString(resourceGroup().id)}', 50)
+  location: location
+  sku: {
+    name: 'Basic'
+  }
+  tags: tags
+}
+
+resource acr_mi_AcrPull 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+  name: guid(acr.id, mi.id, subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '7f951dda-4ed3-4680-a7ca-43fe172d538d'))
+  properties: {
+    principalId: mi.properties.principalId
+    roleDefinitionId: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '7f951dda-4ed3-4680-a7ca-43fe172d538d')
+    principalType: 'ServicePrincipal'
+  }
+  scope: acr
+}
+
+resource acr_mi_AcrPush 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+  name: guid(acr.id, mi.id, subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '8311e382-0749-4cb8-b61a-304f252e45ec'))
+  properties: {
+    principalId: mi.properties.principalId
+    roleDefinitionId: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '8311e382-0749-4cb8-b61a-304f252e45ec')
+    principalType: 'ServicePrincipal'
+  }
+  scope: acr
+}
+
+resource law 'Microsoft.OperationalInsights/workspaces@2023-09-01' = {
+  name: take('law-${uniqueString(resourceGroup().id)}', 63)
+  location: location
+  properties: {
+    sku: {
+      name: 'PerGB2018'
+    }
+  }
+  tags: tags
+}
+
+resource cae 'Microsoft.App/managedEnvironments@2024-03-01' = {
+  name: take('cae${uniqueString(resourceGroup().id)}', 24)
+  location: location
+  properties: {
```

Comment:
Is this up to date? Doing a diff with azd, I see workloadProfiles is missing (left is azd):

![image](https://github.com/user-attachments/assets/0ed5dc94-de3d-4646-90b0-c3ab355bc06f)

But I see code trying to set this.

Reply:
Undid my change.